### PR TITLE
core: Deprecate ForwardingChannelBuilder

### PR DESCRIPTION
**Pull request by sergiitk**
Deprecate `ForwardingChannelBuilder` in favor of `ForwardingChannelBuilder2`

### CODE REVIEW

**Comment by sergiitk** [comment]
cc @larry-safran

File: alts/src/main/java/io/grpc/alts/ComputeEngineChannelBuilder.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
  * using ALTS if applicable and using TLS as fallback.
  */
 public final class ComputeEngineChannelBuilder
-    extends ForwardingChannelBuilder<ComputeEngineChannelBuilder> {
+    extends ForwardingChannelBuilder2<ComputeEngineChannelBuilder> {
```
Comment (ejona86) [inline alts/src/main/java/io/grpc/alts/ComputeEngineChannelBuilder.java:29]: This breaks ABI. We can't change this. Ditto for most of the usages in public APIs. AltsChannelBuilder has few enough users it is probably fine. BinderChannelBuilder would be nice to change, but we may not be easily able to. I'm actually surprised it isn't stable, but I guess it was just BinderServerBuilder that was marked stable.

Comment (larry-safran) [reply alts/src/main/java/io/grpc/alts/ComputeEngineChannelBuilder.java:29]: What about AndroidChannelBuilder?  That also looks to be stabilized.

Are we going to use AltsChannelBuilder2 & AndroidChannelBuilder2?

Comment (sergiitk) [reply alts/src/main/java/io/grpc/alts/ComputeEngineChannelBuilder.java]: Undid:

- AndroidChannelBuilder
- BinderChannelBuilder
- ComputeEngineChannelBuilder
- GoogleDefaultChannelBuilder

**Review by larry-safran — APPROVED** [review]

**Comment by sergiitk** [comment]
Ready for another round

**Comment by sergiitk** [comment]
When it's approved, I'll edit `57d1f8e Fix @SuppressWarnings("deprecation")` description so it doesn't tag the user.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**